adafruit_requests.py

```diff
@@ -37,35 +37,117 @@
 __repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_Requests.git"
 
 import errno
+import sys
+
+if sys.implementation.name == "circuitpython":
+
+    def cast(_t, value):
+        """No-op shim for the typing.cast() function which is not available in CircuitPython."""
+        return value
+
+
+else:
+    from ssl import SSLContext
+    from types import ModuleType, TracebackType
+    from typing import Any, Dict, List, Optional, Tuple, Type, Union, cast
+    from typing_extensions import Protocol
```

Comment:
`Protocol` was first introduced in Python 3.8's `typing` module, so since it seems `adafruit-requests` aims to support Python 3.7 (as seen by the Build CI job steps) and older versions (as seen by the classifiers used in `setup.py`), I added [`typing_extensions`](https://github.com/python/typing/tree/master/typing_extensions) as a dependency to `adafruit_requests` since it provides a backport of `Protocol` for Python versions <3.8.

Reply:
We can move the requirements up to 3.8. The 3.7 choice was made years ago.

Reply:
@dhalbert sure I can do that in this PR.

Are these the changes you would expect? Am I missing anything?

- [ ] Add `python_requires='>=3.8'` to `setup.py`
- [ ] Replace 3.4/3.5 classifiers in `setup.py` with a 3.8 classifier
- [ ] Update `.readthedocs.yaml` to use 3.8 instead of 3.7
- [ ] Update `tox.ini` to use py38 instead of py37
- [ ] Update `build.yml` to use 3.8 instead of 3.7 to run `test` job steps
- [ ] Update `release.yml` to use 3.8 instead of 3.x to run `upload-pypi` job steps
- [ ] Import `Protocol` from `typing` instead of `typing_extensions`
- [ ] Remove `typing-extensions` from `requirements.txt` and `setup.py`'s `install_requires` list

Reply:
@kevincon This list looks good to me. Go ahead and add commit(s) with these changes when you have the opportunity. Thank you for working on this!